keytree/src/lib.rs

```diff
@@ -50,6 +50,36 @@ impl Xprv {
             precompressed_pubkey: self.precompressed_pubkey,
         }
     }
+
+    /// Serializes this Xprv to a sequence of bytes.
+    pub fn to_bytes(&self) -> [u8; 64] {
+        let mut buf = [0u8; 64];
+        buf[..32].copy_from_slice(&self.scalar.to_bytes());
+        buf[32..].copy_from_slice(&self.dk);
+        buf
+    }
+
+    /// Decodes an Xprv from a 64-byte array, and fails if the provided array is not
+    /// exactly 64 bytes.
+    pub fn from_bytes(bytes: &[u8]) -> Option<Self> {
+        if bytes.len() != 64 {
+            return None;
+        }
+
+        let pieces = bytes.split_at(32);
+        let mut scalar_bytes: [u8; 32] = Default::default();
```

Comment:
Arrays have this syntax for initializing with a repeated value:
```rust
        let mut scalar_bytes: [u8; 32] = [0; 32];
```
And if you clarify the integer type, you can even avoid declaring the variable type separately:

```suggestion
        let mut scalar_bytes = [0u8; 32];
```